platforms/avr/led_sysdefs_avr.h

```diff
@@ -52,6 +52,11 @@ extern volatile unsigned long timer0_millis;
 #  endif
 };
 
+// special defs for mega environments
+#if defined(__AVR_ATmega4809__)
+    volatile unsigned long timer0_millis = 0;
+#endif
+
```

Comment:
I'm pretty new to arduino (and haven't done much C for about 25 years!), but I checked out this branch, installed it as a .zip library through the Arduino UI and tried to compile some of the provided example sketches (Blink, etc) for the Nano Every, and every one I tested displayed the following errors:

```
In file included from D:\Users\USER\Documents\Arduino\libraries\FastLED\examples\Blink\Blink.ino:1:0:
D:\Users\USER\Documents\Arduino\libraries\FastLED/FastLED.h:14:21: note: #pragma message: FastLED version 3.003.003
 #    pragma message "FastLED version 3.003.003"
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from D:\Users\USER\Documents\Arduino\libraries\FastLED/FastLED.h:65:0,
                 from D:\Users\USER\Documents\Arduino\libraries\FastLED\examples\Blink\Blink.ino:1:
D:\Users\USER\Documents\Arduino\libraries\FastLED/fastspi.h:135:23: note: #pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output
 #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libraries\FastLED\FastLED.cpp.o (symbol from plugin): In function `CFastLED::CFastLED()':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\bitswap.cpp.o (symbol from plugin): In function `transpose8x1_noinline(unsigned char*, unsigned char*)':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\colorpalettes.cpp.o (symbol from plugin): In function `Rainbow_gp':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\colorutils.cpp.o (symbol from plugin): In function `fill_solid(CRGB*, int, CRGB const&)':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\hsv2rgb.cpp.o (symbol from plugin): In function `hsv2rgb_raw_C(CHSV const&, CRGB&)':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\lib8tion.cpp.o (symbol from plugin): In function `memset8':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\noise.cpp.o (symbol from plugin): In function `inoise16_raw(unsigned long, unsigned long, unsigned long)':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\power_mgt.cpp.o (symbol from plugin): In function `calculate_unscaled_power_mW(CRGB const*, unsigned int)':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
libraries\FastLED\wiring.cpp.o (symbol from plugin): In function `timer0_millis':
(.text+0x0): multiple definition of `timer0_millis'
sketch\Blink.ino.cpp.o (symbol from plugin):(.text+0x0): first defined here
collect2.exe: error: ld returned 1 exit status
exit status 1
Error compiling for board Arduino Nano Every.
```

Commenting out the `#if... #endif` block and inserting the `volatile unsigned long timer0_millis = 0;` line at the top of a sketch just under the `#include <FastLED.h>` line fixes the problem, allowing sketches to compile for Nano Every and run correctly on the device.

I would have registered an issue on the source repo (https://github.com/ngyl88-arduino/FastLED), but it appears to have issues disabled so this was the best place I could find to raise it.

Reply:
Thanks jtq for the feedbacks. I've pushed a commit to remove the redundant definition. Would you be able to try if the compilation works?

I'm new to Arduino too, and didn't really work with C for 10+ years. I don't have any board with me currently, thus wasn't able to verify. There's one thing to mention, I'm not sure on how to verify the pin mapping for RX/TX on Nano Every. See: https://github.com/FastLED/FastLED/issues/716#issuecomment-659010654 . Would you be able to help on that?

Currently the pin definition for `__AVR_ATmega4809__` is based on https://github.com/arduino/ArduinoCore-megaavr/blob/master/cores/arduino/NANO_compat.h , but the RX/TX for Nano Every is different in the `avrmega` working branch https://github.com/FastLED/FastLED/blob/avrmega/platforms/avrmega/fastpin_avrmega.h .

Reply:
Thnaks @ngyl88 - I've created an Issue on your repo (#1 - woo!) so we can better discuss it there.  Thanks for enabling Issues on your repo!

Reply:
Regarding the pin definitions, I'm not sure if this is good enough, but I do have a sketch that sends data out of a digital port to a string of LEDs.

The Nano Every board is labelled TX1, RX0 (, RST, GND, etc, reading from left to right along the board edge).  It does appear that presently a sketch emitting data on digital port "1" in the code is actually sending it out via RX0, and a sketch sending data out digital port "0" is actually emitting it through the TX1 pin... so I'm guessing from that that the NANO_compat.h version is correct, and the work-in-progress avrmega branch is incorrect.

That would also make sense, since I understand a major selling point of the Nano Every is that it's 100% pin-compatible with the older Nano board.

Reply:
To confirm this, a simple test that *doesn't* use FastLED (just sets each digital pin to output HIGH and connects an LED to it) confirms that on the Nano Every digital pin 0 is RX0 (the _second_ pin from the left) and digital pin 1 is TX1 (the _first_ from the left).